handbook/upgrade/upgrade.md

```diff
@@ -0,0 +1,280 @@
+
+    - All servers, file stores, and databases running in virtual machines
+
+    - Removable storage volume or device
+
+## Preparing for Upgrade or Migration
+
+Prior to attempting any upgrade or migration you should backup your server and any remote data stores. Failure to create backups increases the risk of unplanned downtime if issues occur during the upgrade or migration process.
+
+!!!note "Single Node versus Multi Node Configurations"
+    **Single node** refers to the SystemLink Server configuration where the application server, file storage, and databases used by SystemLink are all installed on the same Windows server. This is the default configuration for SystemLink Server.
+
+    **Multi node** refers to configurations where the file store or databases used by SystemLink are running on distinct hardware from the SystemLink application server. This is the recommended configuration for all production deployments.
+
+### NI-SystemLink-Migration Command Line Utility
+
+The workflows for upgrades and migrations make use of the SystemLink command line migration tool, `nislmigrate`. Refer to the readme at the [NI-SystemLink-Migration](https://github.com/ni/NI-SystemLink-Migration) GitHub repo for details on installing and using this tool.
+
+!!!note "Argument Flags for `nislmigrate`"
+    `nislmigrate` supports capturing data from individual services as well as all installed services using the `--all` argument flag. For brevity `--all` is used most workflows in this chapter. Depending on your needs you may replace the `--all` argument flag with one or more of the individual service argument flags.
+
+## Recommended Upgrade and Migrations Workflows for your deployment
+
+This chapter recommends workflows for the following upgrade and migration scenarios. You should plan for some downtime of your SystemLink Server. By following these recommendations this downtime can be minimized.
+
+- [Single Node Upgrade](#single-node-upgrade)
+- [Single Node to Multi Node Migration](#single-node-to-multi-node-migration)
+    - [Single Node to Multi Node with MongoDB](#single-node-to-multi-node-with-mongodb)
+    - [Single Node to Multi Node with File Storage](#single-node-to-multi-node-with-file-storage)
+    - [Single Node to Multi Node with PostgreSQL](#single-node-to-multi-node-with-postgresql)
+- [Upgrading Multi node configurations (MongoDB, PostgreSQL, File Share/S3)](#upgrading-multi-node-configurations)
+- [Seamless cut-over](#seamless-cut-over)
+
+!!!note "Storing Captured Data"
+    For upgrades and migrations for a SystemLink Server with a small amount of data you may be able to use `nislmigrate` to backup your data to the local file system of your SystemLink Server. For SystemLink Servers with a large amount of data be aware `nislmigrate` could exhaust the local storage on the server. This can mitigated by using an attached volume. **This document assumes this will be needed and refers to this attached volume as `D:\` in each workflow.**
```

Comment:
Can we be more specific about what constitutes small/large amounts of data? 

Rewrite for active voice:

If you have a SystemLink Server with a small amount of data, you can use 'nislmigrate' to backup your data to the local file system of your SystemLink Server. For SystemLink Servers with a large amount of data, you may need to use an attached volume since 'nislmigrate' may exhaust the local storage. This document assumes that you are using an attached volume and refers to the volume as 'D:\' in each workflow.

Reply:
I included your recommended wording. Thank you. 

Per small vs large amount of data. This is dependent on how much data they currently have and how much free disk space they have on their server. This can be a bit squishy depending on the scenario and therefore I would rather not define specifically (in GBs or otherwise) what is a small versus large amount of data.